pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdFunctions.java

```diff
@@ -501,13 +511,13 @@ protected void validateFunctionConfigs(FunctionConfig functionConfig) {
                 org.apache.pulsar.common.functions.Utils.inferMissingNamespace(functionConfig);
             }
 
-            if (isNotBlank(functionConfig.getJar()) && isNotBlank(functionConfig.getPy())) {
-                throw new ParameterException("Either a Java jar or a Python file needs to"
+            if (isNotBlank(functionConfig.getJar()) && isNotBlank(functionConfig.getPy()) && isNotBlank(functionConfig.getGo())) {
+                throw new ParameterException("Either a Java jar or a Python file or a Go file needs to"
```

Comment:
yes, in here,  we uploaded a source files suffixed with `.go`.

in [getGoInstanceCmd()](https://github.com/apache/pulsar/pull/4174/files#diff-092ae104654a1e712a083dacc1ee5308R233) 

```
args.add("go");
args.add("run");
args.add(originalCodeFileName);
```

We use go run to run this go file, `go run` can be divided into two steps:

1. build `.go` file is an executable binary
2. run this executable binary

Reply:
@wolfstudy 

Actually I forgot adding the comment there. I don't think we should use "go run", because not every machine installed `go`. so I would prefer uploading the executable binary.

Reply:
awesome, will fix it